src/marshmallow/validate.py

```diff
@@ -32,6 +32,60 @@ def _repr_args(self) -> str:
         """
         return ""
 
+    def __call__(self, value):
+        raise NotImplementedError
+
+
+class And(Validator):
+    """Compose multiple validators and combine their error messages.
+
+    Example: ::
+
+        from marshmallow import validate, ValidationError
+
+        def is_even(value):
+            if value % 2 != 0:
+                raise ValidationError("Not an even value.")
+
+        validator = validate.And([validate.Range(min=0), is_even])
+        validator(-1)
+        # ValidationError: ['Must be greater than or equal to 0.', 'Not an even value.']
+
+    :param validators: Validators to combine.
+    :param error: Error message to use when a validator returns ``False``.
+    """
+
+    default_error_message = "Invalid value."
+
+    def __init__(
+        self,
+        validators: typing.Iterable[typing.Callable[[typing.Any], typing.Any]],
+        *,
+        error: typing.Optional[str] = None
+    ):
+        self.validators = validators
```

Comment:
If a `generator` of `Validator` is passed here, the validation will be called only once at the time of the first pass. I think there should be line like this:
```python
self.validators = tuple(validator)
```